### PR TITLE
Filter template completion items

### PR DESCRIPTION
**Pull request by josa42**
In the atom package [josa42/atom-languageserver-ember](https://github.com/josa42/atom-languageserver-ember), I have the the issue that completion results neither get filtered and nor sorted. This behavior is confusing and breaks the tab completion, because the first item is used, not the best matching one.

This is what it curently looks like.

75afd02c-3ffb-11e7-8de5-aff33c61a1b6

That's what I would expect:

77e5f7a4-3ffb-11e7-928c-cc70aa5722c6

In vscode it works correctly, and I am guessing this is done by the editor. So my first solution was to filter the result in the client (see: atom/atom-languageclient#41).

But on second thought, this should probably be done in the server. So this is my attempt on that. I am using [atom/fuzzaldrin](https://github.com/atom/fuzzaldrin) which is the fuzzy search library used by atom.

Please let me know what you think about it.

### CODE REVIEW

File: src/completion-provider/template-completion-provider.ts [hunk 1]
```diff
@@ -173,3 +174,7 @@ function toCompletionItemKind(type: string): CompletionItemKind {
     return CompletionItemKind.Class;
   }
 }
+
+function getTextPrefix({ node }: ASTPath): string {
+  return node && node.original.replace('ELSCompletionDummy', '');
```
Comment (t-sauer) [inline src/completion-provider/template-completion-provider.ts]: Huh? Why is this necessary? Do our completion items include dummy elements?

Comment (josa42) [inline src/completion-provider/template-completion-provider.ts]: Well in [line L47](https://github.com/josa42/ember-language-server/blob/d1486e92d1682cedd5cde3628ac762e65a30088e/src/completion-provider/template-completion-provider.ts#L47) `ELSCompletionDummy` add, so it needs to be removed here :man_shrugging:

Comment (t-sauer) [inline src/completion-provider/template-completion-provider.ts]: Ah got it now.
